src/pages/workspace/upgrade/UpgradeIntro.tsx

```diff
@@ -33,6 +35,17 @@ function UpgradeIntro({feature, onUpgrade, buttonDisabled, loading, isCategorizi
     const {translate} = useLocalize();
     const {environmentURL} = useEnvironment();
     const subscriptionPlan = useSubscriptionPlan();
+    const preferredCurrency = usePreferredCurrency();
+
+    const formattedPrice = React.useMemo(() => {
+        const upgradeCurrency: keyof typeof CONST.SUBSCRIPTION.PRICES = Object.hasOwn(CONST.SUBSCRIPTION.PRICES, preferredCurrency)
+            ? (preferredCurrency as keyof typeof CONST.SUBSCRIPTION.PRICES)
+            : CONST.PAYMENT_CARD_CURRENCY.USD;
+        const upgradePlan: keyof (typeof CONST.SUBSCRIPTION.PRICES)[typeof upgradeCurrency] =
+            !isCategorizing || subscriptionPlan === CONST.POLICY.TYPE.CORPORATE ? CONST.POLICY.TYPE.CORPORATE : CONST.POLICY.TYPE.TEAM;
+        const annualPrice: number = CONST.SUBSCRIPTION.PRICES[upgradeCurrency][upgradePlan][CONST.SUBSCRIPTION.TYPE.ANNUAL];
+        return `${convertToShortDisplayString(annualPrice, upgradeCurrency)} `;
```

Comment:
I think using ts here seems redundant, the code is simple and there is no any ts error, let's remove it to simplify the code

Reply:
> `|| subscriptionPlan === CONST.POLICY.TYPE.CORPORATE`

I don't understand why you add this new condition? It is not found on the removed way.

Reply:
> I think using ts here seems redundant, the code is simple and there is no any ts error, let's remove it to simplify the code

I get this ts error.

![Screenshot 2024-12-22 at 22 27 22](https://github.com/user-attachments/assets/c020f084-5f47-4dc4-930e-c90cb62f0db8)

> > `|| subscriptionPlan === CONST.POLICY.TYPE.CORPORATE`
> 
> I don't understand why you add this new condition? It is not found on the removed way.

I can probably remove it but it is supposed to be `subscriptionPlan === CONST.POLICY.TYPE.TEAM` so in case `isCategorizing` is `true` we should upgrade to corporate `CONST.POLICY.TYPE.CORPORATE`

Reply:
> I get this ts error.

Hmm! `npm run typecheck` works fine with me.

This is my code

```javascript
const formattedPrice = React.useMemo(() => {
    const upgradeCurrency = Object.hasOwn(CONST.SUBSCRIPTION.PRICES, preferredCurrency) ? preferredCurrency : CONST.PAYMENT_CARD_CURRENCY.USD;
    const upgradePlan = isCategorizing ? CONST.POLICY.TYPE.TEAM : CONST.POLICY.TYPE.CORPORATE;
    const annualPrice = CONST.SUBSCRIPTION.PRICES[upgradeCurrency][upgradePlan][CONST.SUBSCRIPTION.TYPE.ANNUAL];
    return `${convertToShortDisplayString(annualPrice, upgradeCurrency)} `;
}, [preferredCurrency, isCategorizing]);
```

Reply:
> I can probably remove it but it is supposed to be subscriptionPlan === CONST.POLICY.TYPE.TEAM so in case isCategorizing is true we should upgrade to corporate CONST.POLICY.TYPE.CORPORATE

Sorry, I didn't get that. This condition is not used before, can you please elaborate why we need to add it?

Reply:
I get it, I removed it.

Reply:
@ahmedGaber93, It was just a precaution but I believe it's better to remove it because it's unnecessary and it might introduce other regressions I might not be aware of. 😟

If the current subscription plan is `CONST.POLICY.TYPE.TEAM`, it will upgrade the workspace to `CONST.POLICY.TYPE.CORPORATE` even if `isCategorizing` is `true`.

Instead of showing a `CONST.POLICY.TYPE.TEAM` upgrade while it's already a `CONST.POLICY.TYPE.TEAM` workspace.